tests/BenchmarkDotNet.IntegrationTests.ManualRunning/ExpectedBenchmarkResultsTests.cs

```diff
@@ -0,0 +1,220 @@
+            {
+                Assert.True(cpuGhz * report.ResultStatistics.Mean < 1, $"Actual time was greater than 1 clock cycle.");
+
+                var overheadTime = report.AllMeasurements
+                    .Where(m => m.IsOverhead() && m.IterationStage == Engines.IterationStage.Actual)
+                    .Select(m => m.GetAverageTime().Nanoseconds)
+                    .Average();
+
+                var workloadTime = report.AllMeasurements
+                    .Where(m => m.IsWorkload() && m.IterationStage == Engines.IterationStage.Actual)
+                    .Select(m => m.GetAverageTime().Nanoseconds)
+                    .Average();
+
+                // Allow for 1 cpu cycle variance
+                Assert.True(overheadTime * cpuGhz < workloadTime * cpuGhz + 1, "Overhead took more time than workload.");
+
+                Assert.True((report.GcStats.GetBytesAllocatedPerOperation(report.BenchmarkCase) ?? 0L) == 0L, "Memory allocations measured above 0.");
+            }
+        }
+
+        [Fact]
+        public void LargeStructBenchmarksReportsNonZeroTimeAndZeroAllocated_InProcess()
+        {
+            AssertLargeStructResults(ManualConfig.CreateEmpty()
+                .AddJob(Job.Default
+                    .WithToolchain(InProcessEmitToolchain.Instance)
+                ));
+        }
+
+        [TheoryNetCoreOnly("To not repeat tests in both full Framework and Core")]
+        [InlineData(RuntimeMoniker.Net70)]
+        [InlineData(RuntimeMoniker.Mono70)]
+        public void LargeStructBenchmarksReportsNonZeroTimeAndZeroAllocated_Core(RuntimeMoniker runtimeMoniker)
+        {
+            AssertLargeStructResults(ManualConfig.CreateEmpty()
+                .AddJob(Job.Default
+                    .WithRuntime(runtimeMoniker.GetRuntime())
+                ));
+        }
+
+        [TheoryFullFrameworkOnly("Can only run full Framework and Mono tests from Framework host")]
+        [InlineData(RuntimeMoniker.Net462)]
+        [InlineData(RuntimeMoniker.Mono)]
+        public void LargeStructBenchmarksReportsNonZeroTimeAndZeroAllocated_Framework(RuntimeMoniker runtimeMoniker)
+        {
+            AssertLargeStructResults(ManualConfig.CreateEmpty()
+                .AddJob(Job.Default
+                    .WithRuntime(runtimeMoniker.GetRuntime())
+                ));
+        }
+
+        private void AssertLargeStructResults(IConfig config)
+        {
+            var summary = CanExecute<LargeStruct>(config
+                .WithSummaryStyle(SummaryStyle.Default.WithTimeUnit(Perfolizer.Horology.TimeUnit.Nanosecond))
+                .AddDiagnoser(new MemoryDiagnoser(new MemoryDiagnoserConfig(false)))
+            );
+
+            var cpuGhz = RuntimeInformation.GetCpuInfo().MaxFrequency.Value.ToGHz();
+
+            foreach (var report in summary.Reports)
+            {
+                Assert.True(cpuGhz * report.ResultStatistics.Mean >= 1, $"Actual time was less than 1 clock cycle.");
```

Comment:
Please, avoid using the explicit value of the `Mean` when it comes to such sensitive checks. Use `ZeroMeasurementHelper` instead. If this class doesn't fully cover your need, the corresponding API can be extended.

Reply:
Sure. I have no idea how to use that to check if overhead > workload, though.

Reply:
Just use `ZeroMeasurementHelper` the same way it's used in `ZeroMeasurementAnalyser`. I will improve the math and all the relevant usages lately after an update to a newer version of Perfolizer.

Reply:
From what I can tell, it only tells me if the values are equal or not, not whether one value is greater than the other.

Reply:
Also, I just ran a test with that change
```cs
bool isZero = ZeroMeasurementHelper.CheckZeroMeasurementTwoSamples(workloadTimes, overheadTimes);
```
And it came back as zero with overhead 2.5ns and workload 3.3ns (actual 0.77ns). That's 2 clock cycles, so I don't think it should report zero. It seems there is no way to provide a `threshold` to that function.

Reply:
Could you please share the list of raw measurements that produce this result?

Reply:
```
OverheadActual   1: 268435456 op, 670227500.00 ns, 2.4968 ns/op
OverheadActual   2: 268435456 op, 668632800.00 ns, 2.4909 ns/op
OverheadActual   3: 268435456 op, 670826700.00 ns, 2.4990 ns/op
OverheadActual   4: 268435456 op, 679886600.00 ns, 2.5328 ns/op
OverheadActual   5: 268435456 op, 682667300.00 ns, 2.5431 ns/op
OverheadActual   6: 268435456 op, 679714500.00 ns, 2.5321 ns/op
OverheadActual   7: 268435456 op, 683686400.00 ns, 2.5469 ns/op
OverheadActual   8: 268435456 op, 677954000.00 ns, 2.5256 ns/op
OverheadActual   9: 268435456 op, 684013900.00 ns, 2.5482 ns/op
OverheadActual  10: 268435456 op, 687508300.00 ns, 2.5612 ns/op
OverheadActual  11: 268435456 op, 681488300.00 ns, 2.5387 ns/op
OverheadActual  12: 268435456 op, 682893700.00 ns, 2.5440 ns/op
OverheadActual  13: 268435456 op, 679335900.00 ns, 2.5307 ns/op
OverheadActual  14: 268435456 op, 676327200.00 ns, 2.5195 ns/op
OverheadActual  15: 268435456 op, 685008600.00 ns, 2.5519 ns/op

WorkloadActual   1: 268435456 op, 891689200.00 ns, 3.3218 ns/op
WorkloadActual   2: 268435456 op, 889594100.00 ns, 3.3140 ns/op
WorkloadActual   3: 268435456 op, 889453400.00 ns, 3.3135 ns/op
WorkloadActual   4: 268435456 op, 892752800.00 ns, 3.3258 ns/op
WorkloadActual   5: 268435456 op, 889128900.00 ns, 3.3123 ns/op
WorkloadActual   6: 268435456 op, 887634600.00 ns, 3.3067 ns/op
WorkloadActual   7: 268435456 op, 888043100.00 ns, 3.3082 ns/op
WorkloadActual   8: 268435456 op, 888677300.00 ns, 3.3106 ns/op
WorkloadActual   9: 268435456 op, 888590000.00 ns, 3.3103 ns/op
WorkloadActual  10: 268435456 op, 887945000.00 ns, 3.3079 ns/op
WorkloadActual  11: 268435456 op, 888359400.00 ns, 3.3094 ns/op
WorkloadActual  12: 268435456 op, 888814600.00 ns, 3.3111 ns/op
WorkloadActual  13: 268435456 op, 888303000.00 ns, 3.3092 ns/op
WorkloadActual  14: 268435456 op, 887453300.00 ns, 3.3060 ns/op
WorkloadActual  15: 268435456 op, 890899300.00 ns, 3.3189 ns/op

WorkloadResult   1: 268435456 op, 209707500.00 ns, 0.7812 ns/op
WorkloadResult   2: 268435456 op, 209566800.00 ns, 0.7807 ns/op
WorkloadResult   3: 268435456 op, 209242300.00 ns, 0.7795 ns/op
WorkloadResult   4: 268435456 op, 207748000.00 ns, 0.7739 ns/op
WorkloadResult   5: 268435456 op, 208156500.00 ns, 0.7754 ns/op
WorkloadResult   6: 268435456 op, 208790700.00 ns, 0.7778 ns/op
WorkloadResult   7: 268435456 op, 208703400.00 ns, 0.7775 ns/op
WorkloadResult   8: 268435456 op, 208058400.00 ns, 0.7751 ns/op
WorkloadResult   9: 268435456 op, 208472800.00 ns, 0.7766 ns/op
WorkloadResult  10: 268435456 op, 208928000.00 ns, 0.7783 ns/op
WorkloadResult  11: 268435456 op, 208416400.00 ns, 0.7764 ns/op
WorkloadResult  12: 268435456 op, 207566700.00 ns, 0.7732 ns/op
WorkloadResult  13: 268435456 op, 211012700.00 ns, 0.7861 ns/op
```

Reply:
Oh, that was user error... I was taking overhead for both. 🤦

Reply:
Is this how it works? It actually tells if it's greater, despite its name?

```cs
var workloadMeasurements = report.AllMeasurements.Where(m => m.Is(IterationMode.Workload, IterationStage.Actual)).GetStatistics().WithoutOutliers();
var overheadMeasurements = report.AllMeasurements.Where(m => m.Is(IterationMode.Overhead, IterationStage.Actual)).GetStatistics().WithoutOutliers();

bool isZero = ZeroMeasurementHelper.CheckZeroMeasurementTwoSamples(workloadMeasurements, overheadMeasurements);
Assert.True(isZero, $"Actual time was not 0.");

isZero = ZeroMeasurementHelper.CheckZeroMeasurementTwoSamples(overheadMeasurements, workloadMeasurements);
Assert.True(isZero, "Overhead took more time than workload.");
```